o-clock.cabal

```diff
@@ -19,26 +20,30 @@ stability:           stable
 extra-doc-files:     CHANGELOG.md
                    , README.md
                    , README.lhs
-cabal-version:       2.0
 tested-with:         GHC == 8.6.5
                    , GHC == 8.8.3
                    , GHC == 8.10.4
+                   , GHC == 9.0.1
+                   , GHC == 9.2.1
 
 source-repository head
   type:     git
   location: https://github.com/serokell/o-clock
 
+common common-options
+  build-depends:
+      base >=4.12 && <4.17
```

Comment:
Yes, it actually happens quite often. See https://hackage.haskell.org/package/base-4.16.0.0/changelog. For example we can see that in 4.16.0.0 something was removed:
> Remove Data.Semigroup.Option and the accompanying option function.

Btw, it was deprecated in 4.15.0.0 first. A more important (potentially affecting more users) change can be found in 4.13:
> The final phase of the MonadFail proposal has been implemented:

Besides, in practice `base` version sort of determines the compiler version (a set of versions to be precise, since different compiler versions may come with the same `base`) and a new version of compiler may easily have breaking changes.

And since `o-clock` uses some advanced things that appeared in `base` not so long ago (e. g. see https://github.com/serokell/o-clock/commit/a69363c4374443af7835e8dcaa71754f1f180ee4), I think the chances of breaking something due to new `base-4.x` are not very small.

Reply:
I think `base >=4.12 && <4.17` basically means `ghc >= 8.6 && < 9.4` (not exactly true, since 9.4 may have base-4.16 too).

Reply:
Ah makes sense then, thanks!